link-verifier/verify-links.py

```diff
@@ -0,0 +1,396 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import time
+import argparse
+import re
+import subprocess
+import requests
+import shutil
+from bs4 import BeautifulSoup
+from termcolor import cprint
+from multiprocessing import Pool
+import traceback
+
+MARKDOWN_SEARCH_TERM = r'\.md$'
+# Regex to find a URL
+URL_SEARCH_TERM = r'((https?|ftp|file)://[-A-Za-z0-9\+&@#\./%\?=~_|!;,;]*[-A-Za-z0-9\+&@#/%=~_|])'
```

Comment:
The extensive regex string is being used to avoid issues with matching characters beyond the link like period (`.`), closing parenthesis `)` or bracket `]` as we look for links present in HTML hrefs, hyperlinks and code comments.

Reply:
Will the link not be preceded by whitespace?

Reply:
Not necessarily, for example hyperlinks in Mardown are preceded by opening bracket `[`

Reply:
If you wish to avoid matching a character, a negated character class might be cleaner.

Reply:
A negated character class based regex would like something like: 'r'(\b(https?|ftp|file)://[^\s\)\]\\"<>]+`, and may not be catch-all for other characters intentionally or un-intentionally are used in documentation beyond links across our repositories